fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/Correlated.java

```diff
@@ -151,6 +151,26 @@
      */
     boolean semanticEquals(@Nullable Object other, @Nonnull AliasMap aliasMap);
 
+    /**
+     * Static helper to shorten boilerplate for callers that have to deal with nullable objects.
+     * @param one one instance
+     * @param other another instance
+     * @param aliasMap an {@link AliasMap}
+     * @param <S> the type parameter for {@code Correlated}
+     * @return This method returns {@code true} if both objects passed in are {@code null}, {@code false}
+     *         if {@code one} is {@code null} but {@code another} is not and defers to the instance-based semantic
+     *         equality in any other case.
+     */
+    static <S extends Correlated<S>> boolean semanticEquals(@Nullable Correlated<S> one, @Nullable Correlated<S> other, @Nonnull AliasMap aliasMap) {
```

Comment:
package protested?  Should this be public?

Reply:
It's defacto `public` as `Correlated` is an interface.